core/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricIT.java

```diff
@@ -102,42 +224,65 @@ public void setupSuiteScopeCluster() throws Exception {
                     jsonBuilder().startObject().field("value", i * 2).endObject()));
         }
 
+        // When using the MockScriptPlugin we can map Stored scripts to inline scripts:
+        // the id of the stored script is used in test method while the source of the stored script
+        // must match a predefined script from CustomScriptPlugin.pluginScripts() method
         assertAcked(client().admin().cluster().preparePutStoredScript()
-                .setScriptLang(GroovyScriptEngineService.NAME)
-                .setId("initScript_indexed")
+                .setScriptLang(CustomScriptPlugin.NAME)
+                .setId("initScript_stored")
                 .setSource(new BytesArray("{\"script\":\"vars.multiplier = 3\"}")));
 
         assertAcked(client().admin().cluster().preparePutStoredScript()
-                .setScriptLang(GroovyScriptEngineService.NAME)
-                .setId("mapScript_indexed")
+                .setScriptLang(CustomScriptPlugin.NAME)
+                .setId("mapScript_stored")
                 .setSource(new BytesArray("{\"script\":\"_agg.add(vars.multiplier)\"}")));
 
         assertAcked(client().admin().cluster().preparePutStoredScript()
-                .setScriptLang(GroovyScriptEngineService.NAME)
-                .setId("combineScript_indexed")
-                .setSource(new BytesArray("{\"script\":\"newaggregation = []; sum = 0;for (a in _agg) { sum += a}; newaggregation.add(sum); return newaggregation\"}")));
+                .setScriptLang(CustomScriptPlugin.NAME)
+                .setId("combineScript_stored")
+                .setSource(new BytesArray("{\"script\":\"sum agg values as a new aggregation\"}")));
 
         assertAcked(client().admin().cluster().preparePutStoredScript()
-                .setScriptLang(GroovyScriptEngineService.NAME)
-                .setId("reduceScript_indexed")
-                .setSource(new BytesArray("{\"script\":\"newaggregation = []; sum = 0;for (agg in _aggs) { for (a in agg) { sum += a} }; newaggregation.add(sum); return newaggregation\"}")));
+                .setScriptLang(CustomScriptPlugin.NAME)
+                .setId("reduceScript_stored")
+                .setSource(new BytesArray("{\"script\":\"sum aggs of agg values as a new aggregation\"}")));
 
         indexRandom(true, builders);
         ensureSearchable();
     }
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
-        Settings settings = Settings.builder()
+        Path home = createTempDir();
+        Path scripts = home.resolve("config").resolve("scripts");
+
+        try {
+            Files.createDirectories(scripts);
+
+            // When using the MockScriptPlugin we can map File scripts to inline scripts:
+            // the name of the file script is used in test method while the source of the file script
+            // must match a predefined script from CustomScriptPlugin.pluginScripts() method
```

Comment:
Yes. I guess we could mock the file loading too  so that we don't need to create files but I like the fact that `MockeScriptEngine` uses the default stuff and only differs when it comes to script compilation and execution.